core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/util/PromiseMethodReturnValueHandler.java

```diff
@@ -51,23 +54,10 @@ public void handleReturnValue(Object returnValue, MethodParameter returnType, Mo
 			final Promise<?> promise = (Promise<?>) returnValue;
 			final DeferredResult<ResponseEntity<?>> result = new DeferredResult<>();
 			promise
-				.then(body -> {
-					if (result.isSetOrExpired()) {
-						LOG.warn("Deferred result is already set or expired, could not deliver result {}.", body);
-					} else { 
-						final ResponseEntity<?> response;
-						if (body instanceof ResponseEntity<?>) {
-							response = (ResponseEntity<?>) body;
-						} else {
-							response = ResponseEntity.ok().body(body);
-						}
-						result.setResult(response);
-					}
-					return null;
-				})
+				.thenRespond(promiseResponse -> setDeferredResult(result, promiseResponse))
 				.fail(err -> {
 					if (result.isSetOrExpired()) {
-						LOG.warn("Deferred result is already set or expired, could not deliver Throwable.", err);
+						LOG.warn("Deferred result is already set or expired, could not deliver Throwable {}.", err);
```

Comment:
Without the `{}` placeholder in the format string we get the `Throwable`'s message and a stack trace, with it included the output will consist of whatever `Throwable#toString` or its overridden implementation would like to tell us (typically the class name and message, or just the message).

Reply:
Right, removing it.

Reply:
Done